components/rule-mgt/org.wso2.carbon.identity.rule.metadata/src/main/java/org/wso2/carbon/identity/rule/metadata/config/FlowConfig.java

```diff
@@ -90,4 +104,81 @@ private static void validateFlow(String flowName) throws RuleMetadataConfigExcep
+     */
+    private static FieldDefinition resolveFieldDefinition(FieldDefinitionConfig fieldDefinitionConfig,
+                                                          Map<String, Object> entry)
+            throws RuleMetadataConfigException {
+
+        if (!entry.containsKey("fieldName")) {
+            throw new RuleMetadataConfigException("'fieldName' is required for a field");
+        }
+
+        String fieldName = entry.get("fieldName").toString();
+        if (!fieldDefinitionConfig.getFieldDefinitionMap().containsKey(fieldName)) {
+            throw new RuleMetadataConfigException("Invalid field: " + fieldName);
+        }
+
+        return fieldDefinitionConfig.getFieldDefinitionMap().get(fieldName);
+    }
+
+    /**
+     * Get the updated field if the field is overridden.
+     *
+     * @param mapper           ObjectMapper
+     * @param overrides        Overrides
+     * @param fieldDefinition  FieldDefinition
+     * @return Updated field
+     */
+    private static Field getUpdatedField(ObjectMapper mapper, Map<?, ?> overrides, FieldDefinition fieldDefinition) {
+
+        if (!overrides.containsKey("field")) {
+            return fieldDefinition.getField();
+        }
+
+        return mapper.convertValue(overrides.get("field"), Field.class);
+    }
+
+    /**
+     * Get the updated value if the value is overridden.
+     *
+     * @param mapper           ObjectMapper
+     * @param overrides        Overrides
+     * @param fieldDefinition  FieldDefinition
+     * @return Updated value
+     */
+    private static Value getUpdatedValue(ObjectMapper mapper, Map<?, ?> overrides, FieldDefinition fieldDefinition) {
+
+        if (!overrides.containsKey("value")) {
+            return fieldDefinition.getValue();
+        }
+
+        Value initialValue = fieldDefinition.getValue();
+        if (initialValue instanceof OptionsInputValue) {
+            OptionsInputValue overriddenOptionsInputValue = mapper.convertValue(overrides.get("value"),
+                    OptionsInputValue.class);
+            return new OptionsInputValue(initialValue.getValueType(), overriddenOptionsInputValue.getValues());
+        }
+
+        if (initialValue instanceof OptionsReferenceValue) {
+            OptionsReferenceValue initialOptionsReferenceValue = (OptionsReferenceValue) initialValue;
+            OptionsReferenceValue overriddenOptionsReferenceValue = mapper.convertValue(overrides.get("value"),
+                    OptionsReferenceValue.class);
+            return new OptionsReferenceValue.Builder()
+                    .valueReferenceAttribute(initialOptionsReferenceValue.getValueReferenceAttribute())
+                    .valueDisplayAttribute(initialOptionsReferenceValue.getValueDisplayAttribute())
+                    .valueType(initialOptionsReferenceValue.getValueType())
+                    .links(overriddenOptionsReferenceValue.getLinks())
+                    .build();
```

Comment:
Here we have let to override links only. Can we validate that as well and throw an error if something not intended is overriden

Reply:
Fixed